agent/websockets/connection.go

```diff
@@ -252,3 +264,47 @@ func (conn *Connection) ReadServerMessages() ([]interface{}, error) {
 		return nil, nil
 	}
 }
+
+// injectWebsocketMessage injects a shim header value into a single websocket message in-place.
+// Returns a pointer to a new copy of the struct on success.
+func injectWebsocketMessage(msg *message, injectionPath []string, injectionValues map[string]string) (*message, error) {
+	if msg == nil || injectionValues == nil {
+		return nil, errors.New("unexpected nil message")
+	}
+	if injectionValues == nil {
```

Comment:
You're checking if `injectionValues == nil` twice here, so the second branch never happens.

That being said; I don't see a reason to disallow the `injectionValues` parameter being nil. The zero value for that is `nil`, and it's usually good practice to make the zero value be a reasonable thing to pass in.

Reply:
Fixed, thanks!